From dbaryshkov@gmail.com Fri Dec  7 00:07:07 2007
Return-Path: <SRS0=x/UTRF0x=Q6=lists.arm.linux.org.uk=linux-arm-kernel-bounces+openembedded=haerwu.biz@haerwu.biz>
X-Spam-Checker-Version: SpamAssassin 3.1.7 (2006-10-05) on 3073.s.tld.pl
X-Spam-Level: 
X-Spam-Status: No, score=-1.5 required=5.0 tests=BAYES_00,SPF_FAIL 
	autolearn=disabled version=3.1.7
Delivered-To: haerwu.biz-marcin@haerwu.biz
Received: (qmail 27813 invoked by uid 813007); 7 Dec 2007 00:02:29 -0000
Delivered-To: haerwu.biz-openembedded@haerwu.biz
Received: (qmail 27803 invoked by uid 813007); 7 Dec 2007 00:02:28 -0000
X-clamdmail: clamdmail 0.18a
Received: from zeniv.linux.org.uk (195.92.253.2)
  by smtp.host4.kei.pl with SMTP; 7 Dec 2007 00:02:28 -0000
Received: from [2002:4e20:1eda:1:201:3dff:fe00:156] (helo=lists.arm.linux.org.uk)
	by ZenIV.linux.org.uk with esmtpsa (Exim 4.63 #1 (Red Hat Linux))
	id 1J0Q9g-0005Vf-7a; Thu, 06 Dec 2007 23:29:23 +0000
Received: from localhost ([127.0.0.1] helo=lists.arm.linux.org.uk)
	by lists.arm.linux.org.uk with esmtp (Exim 4.50)
	id 1J0Pvo-0000d6-NY; Thu, 06 Dec 2007 23:15:02 +0000
Received: from ug-out-1314.google.com ([66.249.92.171])
	by lists.arm.linux.org.uk with esmtp (Exim 4.50) id 1J0PoR-0000bz-SM
	for linux-arm-kernel@lists.arm.linux.org.uk;
	Thu, 06 Dec 2007 23:10:07 +0000
Received: by ug-out-1314.google.com with SMTP id 29so955850ugc
	for <linux-arm-kernel@lists.arm.linux.org.uk>;
	Thu, 06 Dec 2007 15:07:14 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=gamma;
	h=domainkey-signature:received:received:date:to:subject:message-id:mime-version:content-type:content-disposition:user-agent:from;
	bh=kjQKYxTmbZXtFO0UaAnYmPE6Fk1iyXNHPNOw6H2kP6U=;
	b=vpDrUZqfbWhRrhxvozaHlI9ZC2UwfZvkqIrEeyX5rmqM9rXJTkm1fQQUJvaw43KzvEmxqPXYvpvHQ663HaRRVtN08xB0t2NcDpTm20QUwznzUbze+c4FPTP+86FQbsfhCck86qmrvfBUoDVD1xS2eVR+MxwI63D3rapCtUNpvvo=
DomainKey-Signature: a=rsa-sha1; c=nofws; d=gmail.com; s=gamma;
	h=date:to:subject:message-id:mime-version:content-type:content-disposition:user-agent:from;
	b=D4Cx347bDXn3mf7g77jwn0W3XSXIZh+CiEev1FGhlTeKRZhnM7x1ht4Z+57AiD/dMCrCq8lHqE8Gy0Rv7x7krF2bmUJaiO7HXfxmLWY+msGF2nTymkr1NKsqipQRIVCcEf0ZBKcIrT0dK94QhuP4Ea4Sb/YOfb2YwrJPyNWsISo=
Received: by 10.78.165.16 with SMTP id n16mr3075565hue.1196982433587;
	Thu, 06 Dec 2007 15:07:13 -0800 (PST)
Received: from doriath.ww600.siemens.net ( [91.122.9.34])
	by mx.google.com with ESMTPS id k5sm5631nfh.2007.12.06.15.07.12
	(version=SSLv3 cipher=OTHER); Thu, 06 Dec 2007 15:07:12 -0800 (PST)
Date: Fri, 7 Dec 2007 02:07:07 +0300
To: linux-arm-kernel@lists.arm.linux.org.uk
Message-ID: <20071206230707.GA13639@doriath.ww600.siemens.net>
MIME-Version: 1.0
Content-Type: text/plain;
  charset=us-ascii
Content-Disposition: inline
User-Agent: Mutt/1.5.17 (2007-11-01)
From: Dmitry Baryshkov <dbaryshkov@gmail.com>
Subject: [PATCH 2/2] tosa_udc_use_gpio_vbus.patch
X-BeenThere: linux-arm-kernel@lists.arm.linux.org.uk
X-Mailman-Version: 2.1.5
Precedence: list
List-Id: ARM Linux kernel discussions <linux-arm-kernel.lists.arm.linux.org.uk>
List-Unsubscribe: <http://lists.arm.linux.org.uk/mailman/listinfo/linux-arm-kernel>,
	<mailto:linux-arm-kernel-request@lists.arm.linux.org.uk?subject=unsubscribe>
List-Archive: <http://lists.arm.linux.org.uk/pipermail/linux-arm-kernel>
List-Post: <mailto:linux-arm-kernel@lists.arm.linux.org.uk>
List-Help: <mailto:linux-arm-kernel-request@lists.arm.linux.org.uk?subject=help>
List-Subscribe: <http://lists.arm.linux.org.uk/mailman/listinfo/linux-arm-kernel>,
	<mailto:linux-arm-kernel-request@lists.arm.linux.org.uk?subject=subscribe>
Sender: linux-arm-kernel-bounces@lists.arm.linux.org.uk
Errors-To: linux-arm-kernel-bounces+openembedded=haerwu.biz+openembedded=haerwu.biz@lists.arm.linux.org.uk
X-Length: 4865
Status: R
X-Status: NC
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  

Use gpio_vbus instead of udc_is_connected for udc on tosa.

Signed-off-by: Dmitry Baryshkov <dbaryshkov@gmail.com>

Index: linux-test/arch/arm/mach-pxa/tosa.c
===================================================================
--- linux-test.orig/arch/arm/mach-pxa/tosa.c	2007-12-06 12:43:04.301350179 +0300
+++ linux-test/arch/arm/mach-pxa/tosa.c	2007-12-06 12:47:50.881381650 +0300
@@ -165,15 +165,10 @@ static void tosa_udc_command(int cmd)
 	}
 }
 
-static int tosa_udc_is_connected(void)
-{
-	return ((GPLR(TOSA_GPIO_USB_IN) & GPIO_bit(TOSA_GPIO_USB_IN)) == 0);
-}
-
-
 static struct pxa2xx_udc_mach_info udc_info __initdata = {
 	.udc_command		= tosa_udc_command,
-	.udc_is_connected	= tosa_udc_is_connected,
+	.gpio_vbus		= TOSA_GPIO_USB_IN,
+	.gpio_vbus_inverted	= 1,
 };
 
 /*

-------------------------------------------------------------------
List admin: http://lists.arm.linux.org.uk/mailman/listinfo/linux-arm-kernel
FAQ:        http://www.arm.linux.org.uk/mailinglists/faq.php
Etiquette:  http://www.arm.linux.org.uk/mailinglists/etiquette.php

